src/hotspot/share/runtime/java.cpp

```diff
@@ -504,9 +505,11 @@ void before_exit(JavaThread* thread) {
 
 #if INCLUDE_CDS
   if (DynamicArchive::should_dump_at_vm_exit()) {
-    assert(ArchiveClassesAtExit != NULL, "Must be already set");
+    const char* archive_name =
+       (ArchiveClassesAtExit != nullptr) ? ArchiveClassesAtExit : Arguments::GetSharedDynamicArchivePath();
+    assert(archive_name != NULL, "Must be already set");
```

Comment:
The above change doesn't seem necessary. `ArchiveClassesAtExit` should be already set to non-null if AutoCreateArchive was set but the dynamic archive cannot be mapped.